client/my-sites/themes/theme-preview.jsx

```diff
@@ -2,79 +2,138 @@
  * External dependencies
  */
 import React from 'react';
+import { connect } from 'react-redux';
 import noop from 'lodash/noop';
 
 /**
  * Internal dependencies
  */
-import WebPreview from 'components/web-preview';
 import Button from 'components/button';
-import { getPreviewUrl } from './helpers';
+import QueryTheme from 'components/data/query-theme';
+import { connectOptions } from './theme-options';
+import { getThemeForPreviewData, getTheme, getThemePreviewState } from 'state/themes/selectors';
+import { getPreviewUrl } from 'my-sites/themes/helpers';
+import { localize } from 'i18n-calypso';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { closeThemePreview } from 'state/themes/actions';
+import WebPreview from 'components/web-preview';
 
-export default React.createClass( {
+const ThemePreview = React.createClass( {
 	displayName: 'ThemePreview',
 
 	propTypes: {
 		theme: React.PropTypes.object,
 		showPreview: React.PropTypes.bool,
 		showExternal: React.PropTypes.bool,
 		onClose: React.PropTypes.func,
-		primaryButtonLabel: React.PropTypes.string.isRequired,
-		onPrimaryButtonClick: React.PropTypes.func,
 		getPrimaryButtonHref: React.PropTypes.func,
 		secondaryButtonLabel: React.PropTypes.string,
 		onSecondaryButtonClick: React.PropTypes.func,
```

Comment:
Can't we drop most of these `primary`/`secondary` props now that we receive options objects as props? (Same for `getDefaultProps`.) We could add `propTypes` for `previewData` tho...